### PR TITLE
snap

### PR DESCRIPTION
**Pull request by mikeroyal**
snapcraft.yaml to build a [snap](https://snapcraft.io/). This is the basic snap layout. I still to make a few configurations for networking.

This is just a checklist to guide you. You can remove it safely.

**Checklist:**

-   [x] If you are new to Scapy: I have checked <https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md> (esp. section submitting-pull-requests)
-   [ ] I squashed commits belonging together
-   [ ] I added unit tests or explained why they are not relevant
-   [ ] I executed the regression tests for Python2 and Python3 (using `tox` or, `cd test && ./run_tests_py2, cd test && ./run_tests_py3`)

> brief description what this PR will do, e.g. fixes broken dissection of XXX
A snap package that will install and setup the scapy shell on Debian/Ubuntu systems.
> if required - short explanation why you fixed something in a way that may look more complicated as it actually is

> if required - outline impacts on other parts of the library

fixes #xxx (add issue number here if appropriate, else remove this line)

### CODE REVIEW

**Comment by guedou** [comment]
Thanks for the PR. I thing that the `.config/snap/` directory will be more appropriate. If you can add a section in https://scapy.readthedocs.io/en/latest/installation.html that will be good too.

What do we need to do to submit the `snap`?

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+parts:
+  scapy:
+    source: https://github.com/secdev/scapy.git
+    source-tag: 2.4.2
+    plugin: python
+    python-version: python3
+
+    stage-packages:
+      - python-six
```
Comment (guedou) [inline snap/snapcraft.yaml]: Is this required? Scapy does not depend on this package.

Comment (mikeroyal) [reply snap/snapcraft.yaml]: Hi @guedou, it's not necessary I just add it the package out habit. If you guys do not depend the package I will remove it. :+1:

**Comment by mikeroyal** [comment]
Hi @guedou, I will get to work on the directory change and install documentation.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (gpotter2) [reply snap/snapcraft.yaml]: We actually ship it ourselves to have no dependency. Because it's a one-file dependency we allowed this.

**Comment by guedou** [comment]
I tried to improve this snap, and did not manage to make it work on Ubuntu 19.04, with snapcraft 2.43.1+18.10. Which versions did you use?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+grade: devmode
+base: core18
+
+architectures:  
+  - build-on: amd64
+
+parts:
+  scapy:
+    source: https://github.com/secdev/scapy.git
+    source-tag: 2.4.2
```
Comment (guedou) [inline snap/snapcraft.yaml]: This should be 2.4.3 or master.

**Comment by gpotter2** [comment]
Closed for inactivity

**Pull request closed without merging**